cogs/ReactForRole.py

```diff
@@ -0,0 +1,716 @@
+        if not member:
+            return
+        if not role_id:
+            return
+        role: discord.Role = discord.utils.get(guild.roles, id=role_id)
+        return member, role
+
+    async def parse_emoji_and_role_input_str(self, ctx: commands.Context, input_str: str, remaining_slots: int) -> List[
+        Tuple[Union[discord.Emoji, str], discord.Role]]:
+        rows = input_str.splitlines()
+        arr = []
+        for row in rows:
+            emoji_role = row.split(',')
+            if len(emoji_role) > 2:
+                raise commands.BadArgument("Too many categories/etc on one line.")
+            emoji: Union[discord.Emoji, str] = await self.get_first_emoji_from_str(ctx, emoji_role[0].strip())
+            if not emoji:
+                await ctx.send(f"Yeah, didn't find emoji for `{emoji_role[0]}`")
+                continue
+            role = await commands.RoleConverter().convert(ctx, emoji_role[1].lstrip().rstrip())
+            arr.append((emoji, role))
+            if len(arr) == remaining_slots:
+                return arr
+        return arr
+
+    async def parse_emoji_or_roles_input_str(self, ctx: commands.Context, input_str: str) -> List[
+        Union[discord.Emoji, str, discord.Role]]:
+        rows = input_str.splitlines()
+        arr = []
+        for row in rows:
+            # Try and match it to an raw_emoji first
+            raw_emoji = await self.get_first_emoji_from_str(ctx, row.strip())
+            if not raw_emoji:
+                role = await commands.RoleConverter().convert(ctx, row.strip())
+                if not role:
+                    raise commands.BadArgument("Couldn't convert to a role or emoji.")
+                else:
+                    arr.append(role)
+            else:
+                arr.append(raw_emoji)
+        return arr
+
+    async def prompt_for_input(self, ctx: commands.Context, input_type: str) -> str:
+        await ctx.send(f"Please enter {input_type} so I can progress further. I'll wait 60 seconds, don't worry.")
+        msg, channel = await self.wait_for_message(self.bot, ctx)
+        if not msg:
+            await channel.send("Okay, I'll cancel the command.")
+        else:
+            return msg.content
+
+    async def overwrite_channel_add_reaction_perms(self, guild: discord.Guild, channel: discord.TextChannel):
+        roles: List[discord.Role] = guild.roles
+        overwrite: discord.PermissionOverwrite = discord.PermissionOverwrite()
+        overwrite.update(add_reactions=False)
+        for role in roles:
+            await channel.set_permissions(role, overwrite=overwrite)
+        bot_members: List[discord.Member] = [member for member in guild.members if member.bot]
+        for bot in bot_members:
+            for role in bot.roles:
+                await channel.set_permissions(role, overwrite=None)
+
+    @staticmethod
+    async def wait_for_message(bot: discord.Client, ctx: commands.Context, timeout: float = 60.0) -> Tuple[
```

Comment:
worth making the 60 a constant at the top so we can edit it later if required, some of these messages are quite long which makes me wonder if 120 would be better in the future. 60 is good for now, but not having this hardcoded everywhere and instead using a global/ class constant is better. Make sure to edit any 'I'll wait 60 seconds' strings to use the constant.

Reply:
Currently it's using the param as a default rather than a constant, the message request when called is automatically adjusted to whatever the arg given was.